overlord/devicestate/devicemgr.go

```diff
@@ -462,6 +475,43 @@ func (m *DeviceManager) ensureBootOk() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureInstalled() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if release.OnClassic {
+		return nil
+	}
+
+	if m.ensureInstalledRan {
+		return nil
+	}
+
+	if m.recoveryMode != "install" {
+		return nil
+	}
+
+	var seeded bool
+	err := m.state.Get("seeded", &seeded)
+	if err != nil {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
```

Comment:
I think we want something like:
```
        if m.changeInFlight("create-partitions") {
		return nil
	}
```
here.

Reply:
Adding the change in flight test.

Reply:
I believe this test should be applied to the `install-device` change and not the `create-partitions` task?

Reply:
uh, sorry, yeah, I probably looked at the wrong place (sry! I blame the jetlag :)